myTodo.ts

```diff
@@ -0,0 +1,67 @@
+class TodoList {
+    private todos: Array<any> = [];
+
+    addTodos(name: string,  date: Date, description: string, finished?: boolean): void {
+        if (name === '') {
+            return;
+        }
+        this.todos.push({
+            name: name,
+            date: date,
+            description: description,
+            finished: false
+        });
+    }
+
+    getUnfinishedTasks(): void {
+        const notFinished = this.todos.filter(todo => !todo.finished);
+        console.table(notFinished);
+    }
+
+    modifyTask(index: number, newName: string): void {
+        const notFinished = this.todos.filter(todo => !todo.finished);
+        let anotherName: string;
+        for (let task of notFinished) {
+            if (task.finished || typeof index !== 'number') {
```

Comment:
Ai complicat lucrurile cam tare :)
Uneori metoda cea mai simpla este si cea mai buna. Nu trebuia sa filtrezi neaparat task-urile terminate, nu vad o problema in a modifica numele chiar daca este terminat. Ok, sa zicem ca are sens, urmatorul pas insa nu consider ca este cea mai buna alegere: ce rost are sa verifici daca task-ul este terminat ? (tocmai ai filtrat toate task-urile), de ce sa verifici daca index este numar la fiecare iteratie ? o poti face la inceput, iar daca index nu este numar, zici pas :). De ce sa returnezi "nimic" ? 
Te-as ruga sa refaci putin implementarea, dar de data asta mult mai simplu, sa vedem ce iese. Merci !